### PR TITLE
Add cabinet filter and popups to map

### PR DESCRIPTION
**Pull request by paylakatel**
This PR:
- adds more datasets to the map
- adds a filter for cabinets 
- adds popups to the features

Question: to update the map based on the filters changes I used componentWillRecieveProps in the vision zero map. Looks like that function has been [depreciated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops). I used [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate) instead - is that what I should be using now?

### CODE REVIEW

File: components/Map.js [hunk 1]
```diff
@@ -52,21 +60,149 @@ class Map extends React.Component {
...
+
+      this.map.addLayer({
+        id: 'publicWorksRamps',
+        type: 'circle',
+        source: 'publicWorksRamps-point',
+        paint: {
+          'circle-stroke-width': 1,
+          'circle-stroke-color': '#091F2F',
+          'circle-color': '#c1c1c1',
+          // Make circles larger as the user zooms from 12 to 22.
```
Comment (fionawhim) [inline components/Map.js:92]: 👍

File: components/Map.js [hunk 2]
```diff
@@ -52,21 +60,149 @@ class Map extends React.Component {
...
+        // We create HTML elements for the populated properties.
+        .setHTML(
+          `<div style="min-width: 200px; max-width: 250px">
+        <div>
+            <ul class="dl dl--sm">
+              <li class="dl-i dl-i--b">
+                <div class="dl-d">${properties[0]}</div>
+              </li>        
+              <li class="dl-i dl-i--b">
+                <div class="dl-t">
```
Comment (fionawhim) [inline components/Map.js:160]: Where’s `properties[1]`?

File: components/Map.js [hunk 3]
```diff
@@ -52,21 +60,149 @@ class Map extends React.Component {
...
+                ${properties[2] ? `${properties[2]}<br/>` : ''}
```
Comment (fionawhim) [inline components/Map.js]: This is slightly janky to manually enumerate up to 5. You could take the array of properties, filter out any that are empty, and `join` with`"<br/>"`.

File: components/Map.js [hunk 4]
```diff
@@ -52,21 +60,149 @@ class Map extends React.Component {
...
+
+  // When the cabinet selection changes, we update the map accordingly.
+  componentDidUpdate(prevProps) {
+    if (prevProps.cabinetSelection !== this.props.cabinetSelection) {
+      if (this.props.cabinetSelection == 'All') {
+        this.map.setLayoutProperty('publicWorksRamps', 'visibility', 'visible');
+        this.map.setFilter('budgetFacilities', ['all']);
+        // The public works data is stored in a different layer, but is part of the
+        // 'Streets' cabinet. When 'Streets' is selected, we make the public works
+        // layer visible and turn it off when other cabinets are selected.
```
Comment (fionawhim) [inline components/Map.js:190]: Good comment.

File: components/Map.js (hunk 2 above)
Comment (paylakatel) [reply components/Map.js:160]: Oh! I forgot it.

File: components/Map.js (hunk 3 above)
Comment (paylakatel) [reply components/Map.js]: Good point. Made this better also realized I didn't even need to filter them - can just map through them. Improving styling for these will be in next PR.

**Review by fionawhim — APPROVED** [review]

**Pull request merged**